lib/bounces.js

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const error = require('./error')
+const P = require('./promise')
+
+module.exports = (config, db) => {
+  const configBounces = config.smtp && config.smtp.bounces || {}
+  const BOUNCES_ENABLED = !! configBounces.enabled
+  const MAX_HARD = configBounces.hard && configBounces.hard.max || 0
+  const MAX_SOFT = configBounces.soft && configBounces.soft.max || 0
+  const MAX_COMPLAINT = configBounces.complaint && configBounces.complaint.max || 0
+  const DURATION_HARD = configBounces.hard && configBounces.hard.duration || Infinity
+  const DURATION_SOFT = configBounces.soft && configBounces.soft.duration || Infinity
+  const DURATION_COMPLAINT = configBounces.complaint && configBounces.complaint.duration || Infinity
+  const BOUNCE_TYPE_HARD = 1
+  const BOUNCE_TYPE_SOFT = 2
+  const BOUNCE_TYPE_COMPLAINT = 3
+
+  const freeze = Object.freeze
+  const BOUNCE_RULES = freeze({
+    [BOUNCE_TYPE_HARD]: freeze({
```

Comment:
Sorry, didn't realize that this new feature may be more obscure, I've known about it for a while and we can finally use it in nodejs v4. It is indeed [Computed property names](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names).

The reason was I already had the key in a constant, and instead of writing the key a second time (and possibly mistyping), or creating a blank object and filling the properties the old way (`obj[name] = bar`), I remembered that with node4 we can just do it this way!